src/magic_spell_effect.cpp

```diff
@@ -503,3 +509,245 @@ void spell_effect::translocate( const spell &sp, const Creature &caster,
+                } else {
+                    ydir = 0;
+                }
+            }
+        }
+    }
+}
+
+void spell_effect::fire_ball( const tripoint &target )
+{
+    explosion_handler::explosion( target, 180, 0.5, true );
+}
+
+void spell_effect::map( const spell &sp, const tripoint &target )
+{
+    player *p = g->critter_at<player>( target );
+    const bool new_map = overmap_buffer.reveal(
+                             point( target.x, target.y ), sp.aoe(), target.z );
+    if( new_map ) {
+        p->add_msg_if_player( m_warning, _( "You have a vision of the surrounding area..." ) );
+        p->mod_moves( -to_moves<int>( 1_seconds ) );
+    }
+}
+
+void spell_effect::blood( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    bool blood = false;
+    for( const tripoint &tmp : g->m.points_in_radius( target, sp.aoe() ) ) {
+        blood |= sp.create_field( tmp ) && g->u.sees( tmp );
+    }
+    if( blood ) {
+        caster.add_msg_if_player( m_warning, _( "Blood soaks out of the ground and walls." ) );
+    }
+}
+
+void spell_effect::fatigue( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    caster.add_msg_if_player( m_warning, _( "The fabric of space seems to decay." ) );
+    int x = rng( target.x - 3, target.x + 3 ), y = rng( target.y - 3, target.y + 3 );
+    sp.create_field( {x, y, target.z} );
+}
+
+void spell_effect::pulse( const spell &sp, const tripoint &target )
+{
+    sounds::sound( target, 30, sounds::sound_t::combat, _( "The earth shakes!" ), true, "misc",
+                   "earthquake" );
+    for( const tripoint &pt : g->m.points_in_radius( target, sp.aoe() ) ) {
+        g->m.bash( pt, sp.damage() );
+        g->m.bash( pt, sp.damage() );  // Multibash effect, so that doors &c will fall
+        g->m.bash( pt, sp.damage() );
+        if( g->m.is_bashable( pt ) && rng( 1, 10 ) >= 3 ) {
+            g->m.bash( pt, 999, false, true );
+        }
+    }
+}
+
+void spell_effect::entrance( const spell &sp, const tripoint &target )
+{
+    for( const tripoint &dest : g->m.points_in_radius( target, sp.aoe() ) ) {
+        monster *const mon = g->critter_at<monster>( dest, true );
+        if( mon && mon->friendly == 0 && rng( 0, 600 ) > mon->get_hp() ) {
+            mon->make_friendly();
+        }
+    }
+}
+
+void spell_effect::bugs( const spell &sp, const Creature &caster, const tripoint &target )
+{
+    int roll = rng( 1, 10 );
+    mtype_id bug = mtype_id::NULL_ID();
+    int num = 0;
+    std::vector<tripoint> empty;
+    for( const tripoint &dest : g->m.points_in_radius( target, sp.aoe() ) ) {
+        if( g->is_empty( dest ) ) {
+            empty.push_back( dest );
+        }
+    }
+    if( empty.empty() || roll <= 4 ) {
+        caster.add_msg_if_player( m_warning, _( "Flies buzz around you." ) );
+    } else if( roll <= 7 ) {
+        caster.add_msg_if_player( m_warning, _( "Giant flies appear!" ) );
+        bug = static_cast<mtype_id>( "mon_fly" );
```

Comment:
```suggestion
        bug = mtype_id( "mon_fly" );
```
`static_cast` is possible here, but it's rather unusual. Same for the code below.

Reply:
There was a commit to master changing many casts to static, so i thought it was better.

Reply:
I assume you refer to a commit that converts C-style casts (like `int i = (int) value`). Those are different thing. Here you create a new `mtype_id` from a string and we usually use this style for it. (But `static_cast` is technically also correct.)

Reply:
i can't find it anymore, so ill use C-style.

Reply:
This is probably what you're talking about <https://github.com/CleverRaven/Cataclysm-DDA/pull/31361>